org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseInitHandler.java

```diff
@@ -114,6 +117,14 @@ public InitializeResult initialize(InitializeParams param) {
 			Preferences prefs = Preferences.createFrom((Map<String, Object>) settings);
 			prefs.setRootPaths(rootPaths);
 			preferenceManager.update(prefs);
+			Hashtable<String, String> eclipseOptions = JavaCore.getOptions();
+			int tabSize = prefs.getTabSize();
+			if (tabSize > 0) {
+				eclipseOptions.put(DefaultCodeFormatterConstants.FORMATTER_TAB_SIZE, String.valueOf(tabSize));
+			}
+			boolean insertSpaces = prefs.isInsertSpaces();
+			eclipseOptions.put(DefaultCodeFormatterConstants.FORMATTER_TAB_CHAR, insertSpaces ? JavaCore.SPACE : JavaCore.TAB);
+			JavaCore.setOptions(eclipseOptions);
```

Comment:
can we move preference settings to the preference manager?

Reply:
you'd expect the settings to be updated after the server starts (onDidChangeConfiguration)

Reply:
Fixed.